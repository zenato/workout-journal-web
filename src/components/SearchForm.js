import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form'
import classNames from 'classnames/bind';
import { Button } from 'components/form';
import styles from './SearchForm.scss';

const cx = classNames.bind(styles);

class SearchForm extends Component {
  render() {
    const { handleSubmit, placeholder } = this.props;
    return (
      <section>
        <form onSubmit={handleSubmit} className={cx('search-form')}>
          <Field name="name" type="text" placeholder={placeholder} component="input" className={cx('form-control')} />
          <Button type="submit" value="Search" />
        </form>
      </section>
    );
  }
}

SearchForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
};

SearchForm.defaultProps = {
  placeholder: 'Input search text.',
};

export default SearchForm;
