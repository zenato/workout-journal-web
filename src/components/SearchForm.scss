@import 'common';

.search-form {
  display: flex;
  margin-top: 10px;
  margin-bottom: 10px;

  .field {
    flex: 1;
    margin-right: 3px;
  }
}

