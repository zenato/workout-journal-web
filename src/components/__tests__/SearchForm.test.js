import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from '../SearchForm';

it('Render without error.', () => {
  const component = shallow(<SearchForm handleSubmit={() => {}} />);
  expect(component).toMatchSnapshot();
});
