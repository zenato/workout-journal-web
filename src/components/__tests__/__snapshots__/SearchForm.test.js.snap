// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render without error. 1`] = `
<section>
  <form
    className="search-form"
    onSubmit={[Function]}
  >
    <Field
      className="form-control"
      component="input"
      name="name"
      placeholder="Input search text."
      type="text"
    />
    <Button
      type="submit"
      value="Search"
    />
  </form>
</section>
`;
